src/webgpu/util/color_space_conversion.ts

```diff
@@ -0,0 +1,195 @@
+import { assert } from '../../common/util/util.js';
+
+import { multiplyMatrices } from './math.js';
+
+/**
+ * The color space conversion function definations are fully aign with
+ * CSS Color Module Level 4 Sample Code: https://drafts.csswg.org/css-color/#color-conversion-code
+ */
+
+/**
+ *  Sample code for color conversions
+ * Conversion can also be done using ICC profiles and a Color Management System
+ * For clarity, a library is used for matrix multiplication (multiply-matrices.js)
+ */
+
+// sRGB-related functions
+
+/**
+ * convert an array of sRGB values
+ * where in-gamut values are in the range [0 - 1]
+ * to linear light (un-companded) form.
+ * https://en.wikipedia.org/wiki/SRGB
+ * Extended transfer function:
+ * for negative values,  linear portion is extended on reflection of axis,
+ * then reflected power function is used.
+ */
+function lin_sRGB(RGB: Array<number>) {
```

Comment:
nit: The name tells you what the output is, but isn't clear about the input. How about `extendedGammaCompressArray`, or just `extendedGammaCompress` and use `.map` with it later? Also then you don't need lin_P3/gam_P3 too.

nit: `RGB` should be lowercase since it's a local variable